### PR TITLE
Migrate kubelet/preemption and kubelet/logs to structured logging

### PR DESCRIPTION
**Pull request by qingwave**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Migrate  `pkg/kubelet/preemption/preemption.go` and `pkg/kubelet/logs/container_log_manager.go` to structured logging

#### Which issue(s) this PR fixes:
Ref #98976

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs

```

### CODE REVIEW

File: pkg/kubelet/logs/container_log_manager.go [hunk 1]
```diff
@@ -412,8 +412,9 @@ func (c *containerLogManager) rotateLatestLog(id, log string) error {
 			// This shouldn't happen.
 			// Report an error if this happens, because we will lose original
 			// log.
-			klog.Errorf("Failed to rename rotated log %q back to %q: %v, reopen container log error: %v", rotated, log, renameErr, err)
+			klog.ErrorS(renameErr, "Failed to rename rotated log %q back to %q: %v", "rotatedLog", rotated, "log", log)
 		}
+		klog.ErrorS(err, "Failed to reopen container log", "containerID", id)
```
Comment (ehashman) [inline pkg/kubelet/logs/container_log_manager.go:417]:
```suggestion
```

This line is not necessary, the error below is being returned as opposed to logged.

File: pkg/kubelet/logs/container_log_manager.go [hunk 2]
```diff
@@ -412,8 +412,9 @@ func (c *containerLogManager) rotateLatestLog(id, log string) error {
 			// This shouldn't happen.
 			// Report an error if this happens, because we will lose original
 			// log.
-			klog.Errorf("Failed to rename rotated log %q back to %q: %v, reopen container log error: %v", rotated, log, renameErr, err)
+			klog.ErrorS(renameErr, "Failed to rename rotated log %q back to %q: %v", "rotatedLog", rotated, "log", log)
```
Comment (ehashman) [inline pkg/kubelet/logs/container_log_manager.go:415]: There are actually two errors here and we've dropped `err`. @serathius what should we do here?

File: pkg/kubelet/preemption/preemption.go [hunk 3]
```diff
@@ -107,7 +108,7 @@ func (c *CriticalPodAdmissionHandler) evictPodsToFreeRequests(admitPod *v1.Pod,
 		// this is a blocking call and should only return when the pod and its containers are killed.
 		err := c.killPodFunc(pod, status, nil)
 		if err != nil {
-			klog.Warningf("preemption: pod %s failed to evict %v", format.Pod(pod), err)
+			klog.ErrorS(err, "Failed to evict pod", "pod", format.Pod(pod))
```
Comment (ehashman) [inline pkg/kubelet/preemption/preemption.go:111]:
```suggestion
			klog.ErrorS(err, "Failed to evict pod", "pod", klog.KObj(pod))
```

File: pkg/kubelet/preemption/preemption.go [hunk 4]
```diff
@@ -116,7 +117,7 @@ func (c *CriticalPodAdmissionHandler) evictPodsToFreeRequests(admitPod *v1.Pod,
 		} else {
 			metrics.Preemptions.WithLabelValues("").Inc()
 		}
-		klog.Infof("preemption: pod %s evicted successfully", format.Pod(pod))
+		klog.InfoS("Pod evicted successfully", "pod", format.Pod(pod))
```
Comment (ehashman) [inline pkg/kubelet/preemption/preemption.go:120]:
```suggestion
		klog.InfoS("Pod evicted successfully", "pod", klog.KObj(pod))
```

**Review by ehashman — CHANGES_REQUESTED** [review]
/ok-to-test
/priority important-soon
/triage accepted

Please update your release note as well. No calls to `Fatal` were replaced, so there is no Action Required.

**Comment by ehashman** [comment]
/hold

until release note unnecessary ACTION REQUIRED gets fixed

**Comment by qingwave** [comment]
> /hold
> 
> until release note unnecessary ACTION REQUIRED gets fixed

@ehashman fiexed as review

File: pkg/kubelet/logs/container_log_manager.go (hunk 2 above)
Comment (serathius) [reply pkg/kubelet/logs/container_log_manager.go]: Looking into next line with return we just wrap the `renameErr`, based on that I think it's ok to report here only the renameErr.

Alternative would be to also change the return to create error that wraps both `err` and `renameErr` and then pass both to klog.ErrorS and return.

**Review by ehashman — COMMENTED** [review]
/lgtm

**Comment by qingwave** [comment]
/assign @mrunalp 
Needs approve label

**Review by mrunalp — APPROVED** [review]

**Pull request merged**